powerstore/datasource_host_group.go

```diff
@@ -0,0 +1,267 @@
+		Description: "HostGroup DataSource.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description:         "Unique identifier of the host group.",
+				MarkdownDescription: "Unique identifier of the host group.",
+				Optional:            true,
+				Computed:            true,
+				Validators: []validator.String{
+					stringvalidator.ConflictsWith(path.MatchRoot("name")),
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description:         "Host group name.",
+				MarkdownDescription: "Host group name.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+
+			"host_groups": schema.ListNestedAttribute{
+				Description:         "List of host groups.",
+				MarkdownDescription: "List of host groups.",
+				Computed:            true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"id": schema.StringAttribute{
+							Description:         "Unique identifier of the host group.",
+							MarkdownDescription: "Unique identifier of the host group.",
+							Computed:            true,
+						},
+						"name": schema.StringAttribute{
+							Description:         "Host group name.",
+							MarkdownDescription: "Host group name.",
+							Computed:            true,
+						},
+						"description": schema.StringAttribute{
+							Description:         "Host group description.",
+							MarkdownDescription: "Host group description.",
+							Computed:            true,
+						},
+						"host_connectivity": schema.StringAttribute{
+							Description:         "Connectivity type for hosts and host groups.",
+							MarkdownDescription: "Connectivity type for hosts and host groups.",
+							Computed:            true,
+						},
+						"host_connectivity_l10n": schema.StringAttribute{
+							Description:         "Localized message string corresponding to host_connectivity",
+							MarkdownDescription: "Localized message string corresponding to host_connectivity",
+							Computed:            true,
+						},
+						"hosts": schema.ListNestedAttribute{
+							Description:         "Properties of a host.",
+							MarkdownDescription: "Properties of a host.",
+							Computed:            true,
+							NestedObject: schema.NestedAttributeObject{
+								Attributes: map[string]schema.Attribute{
+									"id": schema.StringAttribute{
+										Description:         "Unique identifier of the host.",
+										MarkdownDescription: "Unique identifier of the host.",
+										Computed:            true,
+									},
+									"name": schema.StringAttribute{
+										Description:         "The host name.",
+										MarkdownDescription: "The host name.",
+										Computed:            true,
+									},
+									"host_group_id": schema.StringAttribute{
+										Description:         "Associated host group, if host is part of host group.",
```

Comment:
I think its replicated field. Anyway the host that will be displayed here are part of hostgroup only